docsbox/docs/views.py

```diff
@@ -31,9 +29,22 @@ def get(self, task_id):
         else:
             return abort(404, message="Unknown task_id")
 
-
 class DocumentTypeView(Resource):
 
+    def post(self):
+        """
+        Returns the File Mimetype of the given file
+        """
+        if "file" in request.files:
+            tmp_file = create_temp_file(request.files["file"], False)
+            mimetype = get_file_mimetype(tmp_file)
+            isConvertable = mimetype not in app.config["ACCEPTED_MIMETYPES"] and mimetype in app.config["CONVERTABLE_MIMETYPES"]
+            return { "convertable": isConvertable, "fileType": mimetype }
+        else:
+            return abort(400, message="file field is required")
+
+class DocumentTypeFromIdView(Resource):        
```

Comment:
Do these need to be separate functions?

Take in the file and the id, if there is an id, get it from VIA, if not, use the file - if neither -> return 400.

Same goes for the other functions.

Reply:
Fixed - I've joined the functions